sound/hda/hdac_stream.c

```diff
@@ -448,13 +457,11 @@ int snd_hdac_stream_setup_periods(struct hdac_stream *azx_dev)
 
 	for (i = 0; i < periods; i++) {
 		if (i == periods - 1 && pos_adj)
-			ofs = setup_bdle(bus, snd_pcm_get_dma_buf(substream),
-					 azx_dev, &bdl, ofs,
-					 period_bytes - pos_adj, 0);
+			ofs = setup_bdle(bus, dma_buffer_p, azx_dev,
+					 &bdl, ofs, period_bytes - pos_adj, 0);
 		else
-			ofs = setup_bdle(bus, snd_pcm_get_dma_buf(substream),
-					 azx_dev, &bdl, ofs,
-					 period_bytes,
+			ofs = setup_bdle(bus, dma_buffer_p, azx_dev,
+					 &bdl, ofs, period_bytes,
```

Comment:
same here, this should be an initial patch, and later on you add support for compressed streams.

Reply:
Separated "preparation changes" in v6 with commit:
ALSA: hda: Prepare for compress stream support